### ISSUE TITLE
Error: IndexError: list index out of range while running policy on us-west-2

### ISSUE THREAD

**Issue by victorh3** [capitalone/cloud-custodian#1005]
My policy is as follows:

policies:
  - name: ec2-report
    resource: ec2
    query:
      - instance-state-name: stopped
    filters:
      - "tag:maid_status": absent
      - type: state-age
        op: gt
        days: 29
    actions:
      - type: mark-for-op
        op: terminate
        days: 3

Seems to work fine when running in US-EAST-1, but i get the following error on US-WEST-2:

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/c7n/policy.py", line 211, in run
    resources = self.policy.resource_manager.resources()
  File "/usr/lib/python2.7/site-packages/c7n/resources/ec2.py", line 89, in resources
    return super(EC2, self).resources(query=query)
  File "/usr/lib/python2.7/site-packages/c7n/query.py", line 284, in resources
    return self.filter_resources(resources)
  File "/usr/lib/python2.7/site-packages/c7n/manager.py", line 86, in filter_resources
    resources = f.process(resources, event)
  File "/usr/lib/python2.7/site-packages/c7n/filters/core.py", line 157, in process
    return filter(self, resources)
  File "/usr/lib/python2.7/site-packages/c7n/filters/core.py", line 407, in __call__
    v = self.get_resource_date(i)
  File "/usr/lib/python2.7/site-packages/c7n/resources/ec2.py", line 200, in get_resource_date
    return parse(self.RE_PARSE_AGE.findall(v)[0][1:-1])
IndexError: list index out of range
2017-03-03 12:17:55,516: custodian.commands:ERROR Error while executing policy ec2-report, continuing
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/c7n/commands.py", line 173, in run
    policy()
  File "/usr/lib/python2.7/site-packages/c7n/policy.py", line 569, in __call__
    return PullMode(self).run()
  File "/usr/lib/python2.7/site-packages/c7n/policy.py", line 211, in run
    resources = self.policy.resource_manager.resources()
  File "/usr/lib/python2.7/site-packages/c7n/resources/ec2.py", line 89, in resources
    return super(EC2, self).resources(query=query)
  File "/usr/lib/python2.7/site-packages/c7n/query.py", line 284, in resources
    return self.filter_resources(resources)
  File "/usr/lib/python2.7/site-packages/c7n/manager.py", line 86, in filter_resources
    resources = f.process(resources, event)
  File "/usr/lib/python2.7/site-packages/c7n/filters/core.py", line 157, in process
    return filter(self, resources)
  File "/usr/lib/python2.7/site-packages/c7n/filters/core.py", line 407, in __call__
    v = self.get_resource_date(i)
  File "/usr/lib/python2.7/site-packages/c7n/resources/ec2.py", line 200, in get_resource_date
    return parse(self.RE_PARSE_AGE.findall(v)[0][1:-1])
IndexError: list index out of range

Any idea what is causing this?
When I run based on instance-uptime, I have no errors in either region.

**Comment by kapilt** [comment]
looks like a value for state transition reason that we haven't seen before. any chance you could report unique values for the StateTransitionReason that aren't user initiated or blank .. via
```aws ec2 describe-instances --query Reservations[].Instances[].StateTransitionReason```

**Comment by victorh3** [comment]
Running the command returns:

[
    "User initiated (2017-02-01 00:30:38 GMT)",
    "",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "",
    "",
    "",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "",
    "",
    "",
    "User initiated (2017-03-06 00:30:39 GMT)",
    "",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "",
    "User initiated (2017-03-02 00:30:41 GMT)",
    "",
    "User initiated (2017-03-03 20:18:03 GMT)",
    "",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "User initiated (2017-02-23 00:30:39 GMT)",
    "",
    "",
    "",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "User initiated (2017-03-01 00:30:39 GMT)",
    "",
    "",
    "User initiated (2017-03-03 20:18:03 GMT)",
    "",
    "User initiated (2017-02-16 00:30:39 GMT)",
    "User initiated (2017-02-18 00:30:40 GMT)",
    "User initiated (2017-02-01 00:30:38 GMT)",
    "User initiated (2017-02-23 00:30:39 GMT)",
    "",
    "User initiated (2017-03-02 00:30:41 GMT)",
    "User initiated (2016-09-15 00:32:26 GMT)"
]

**Comment by kapilt** [comment]
thanks

**Comment by scotwk** [comment]
@kapilt - just saw this now.  I will take a look at it.

(I have to figure out how to handle notifications better on github.  I get notified of everything that happens on cloud custodian, so the real ones get lost in the noise).

**Comment by scotwk** [comment]
@victorh3 - what version of custodian are you running?

PR #945 fixed problems related to parsing dates, which was released in 0.8.23.1 on March 1st.

The new code looks like this:

```
        dates = self.RE_PARSE_AGE.findall(v)
        if dates:
            return parse(dates[0][1:-1])
```

You can see that is different than your stack trace which shows the old code:

```
return parse(self.RE_PARSE_AGE.findall(v)[0][1:-1])
```

If I am correct that you are running an old version, upgrading to the new version should make this problem go away.

Thanks,
scot

**Comment by scotwk** [comment]
@victorh3 - I'm going to close this because I think it is resolved.  If you still have questions please comment and reopen.
